packages/protocol/migrations_sol/deploy_libraries.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Name of temporary directory
+TEMP_DIR_NAME=".tmp/libraries"
+TEMP_DIR="$PWD/$TEMP_DIR_NAME"
+
+# Create a temporary directory or remove it first it if exists
+if [ -d "$TEMP_DIR" ]; then
+    echo "Removing existing temporary folder..."
+    rm -rf $TEMP_DIR
+fi
+mkdir $TEMP_DIR
+
+# Copy libraries to the directory
+LIBRARIES_PATH=("contracts/common/linkedlists/AddressSortedLinkedListWithMedian.sol:AddressSortedLinkedListWithMedian"
+                "contracts/common/Signatures.sol:Signatures"
+                "contracts/common/linkedlists/AddressLinkedList.sol:AddressLinkedList"
+                "contracts/common/linkedlists/AddressSortedLinkedList.sol:AddressSortedLinkedList"
+                "contracts/common/linkedlists/IntegerSortedLinkedList.sol:IntegerSortedLinkedList"
+                "contracts/governance/Proposals.sol:Proposals"
+)
+
+for LIB_PATH in "${LIBRARIES_PATH[@]}"; do
+    IFS=":" read -r SOURCE DEST <<< "$LIB_PATH"
+    DEST_FILE="$TEMP_DIR/$SOURCE"
+    DEST_DIR=$(dirname "$DEST_FILE")
+    mkdir -p "$DEST_DIR"
+    echo "Copying file $SOURCE to $DEST_FILE"
+    cp "$SOURCE" "$DEST_FILE"
+done
+
+# Copy dependencies of the libraries to the directory
+LIBRARY_DEPENDENCIES_PATH=(
+    "contracts/common/FixidityLib.sol"
+    "contracts/common/linkedlists/LinkedList.sol"
+    "contracts/common/linkedlists/SortedLinkedList.sol"
+    "contracts/common/linkedlists/SortedLinkedListWithMedian.sol"
+    "lib/openzeppelin-contracts/contracts/math/SafeMath.sol"
+    "lib/openzeppelin-contracts/contracts/math/Math.sol"
+    "lib/openzeppelin-contracts/contracts/cryptography/ECDSA.sol"   
+    "lib/openzeppelin-contracts/contracts/utils/Address.sol"
+    "lib/solidity-bytes-utils/contracts/BytesLib.sol"
+)
+
+# Creating two variables for better readability
+SOURCE_DIR=$PWD
+DEST_DIR=$TEMP_DIR
+
+for LIB_PATH in "${LIBRARY_DEPENDENCIES_PATH[@]}"; do
+    # Creates directory for the dependency, including any necessary parent directories
+    mkdir -p $DEST_DIR/$(dirname $LIB_PATH)
+    # Copies dependency to the newly created directory
+    cp $SOURCE_DIR/$LIB_PATH $DEST_DIR/$LIB_PATH
+done
+
+# Copy foundry config to the temporary directory
+# TODO(Arthur): Check if using --config-path $SOURCE_DIR/foundry.toml works instead of copying file
```

Comment:
Nit: I'll briefly look into this and address or remove the `TODO` comment. 
Any changes will make CI run again, but shouldn't significantly change the logic of the script.

Reply:
I tried something like this, but Foundry didn't seem to only compile the contracts I wanted. 
Instead, it compiled all contracts.

```sh
forge build --root $TEMP_DIR --contracts $TEMP_DIR --config-path $SOURCE_DIR/foundry.toml
```

I'm ok with the current `pushd`/`popd` solution.

Reply:
Removed comment in https://github.com/celo-org/celo-monorepo/pull/11003/commits/6b66cd3af95847bfc5ace0598d011148562e20d6